### ISSUE TITLE
Detect "user_type" of marketplace proposal

### ISSUE THREAD

**Issue by Waldz** [mysteriumnetwork/node#836]
- [ ] Investigate DB providers & buy IP2Country database 
- [ ] Update DB in our `node` location detection
- [ ] Implement "user_type" field in Discovery service (datacenter/residential)
- [ ] Implement "user_type" field in Dashboard service (datacenter/residential)
- [ ] Node autodetects "user_type" same ways as "country" and adds to Proposal

DB providers:
- https://www.maxmind.com/en/solutions/geoip2-enterprise-product-suite/anonymous-ip-database (free DB version already integrated in our node https://github.com/mysteriumnetwork/node/tree/master/core/location/db)
- https://www.xmyip.com/
- http://getipintel.net/

**Comment by Waldz** [comment]
Maxmind details:
- https://github.com/mysteriumnetwork/node/tree/master/core/location/db)
- https://dev.maxmind.com/faq/should-i-choose-the-geoip-web-service-or-the-database/

**Comment by zolia** [comment]
`GET /location/me` is it really a MORQA endpoint? How it would work? I would say it can be a tequila endpoint.

It should be enough `/location/1.2.3.4` for MORQA.

**Comment by Waldz** [comment]
It can be 1 request, yes. I still suggest to introduce end-point `GET /me` if somebody needs just IP autodetection

**Comment by Waldz** [comment]
Maybe it's logic to implement new Location Service for those endpoints

**Comment by vkuznecovas** [comment]
Please expand on  `Introduce new MORQA endpoints for location metrics`. Absolutely not clear what you mean there.

**Comment by Waldz** [comment]
End-point descriptions with exact JSON structures are provided. 
Will MaxMind documentation to assignee who will be doing it

**Comment by soffokl** [comment]
> Node autodetects "user_type" same ways as "country" and adds to Proposal
> Possible to have manual provider also:

If we are adding `user_type` into the proposal and allowing manual setting it, it can be easy faked by the provider. 

If the `user_type` is important to the consumer - consumer should send a request to detect a reliable type of provider. 

Do we really need this field in the proposal if we cannot trust it?

**Comment by soffokl** [comment]
> Node has several IP detection providers:
> Node has several location detection providers:

Should we support several services for redundancy? If one failed we should ask another - in this case, we need to have some priorities. 

Or is it just a configuration option to choose one at the start?

**Comment by soffokl** [comment]
And it's not clear what is a difference between IP detection and location detection, location metrics?

**Comment by soffokl** [comment]
> Introduce new MORQA endpoints for IP detection
> Introduce new MORQA endpoints for location metrics

Should it be MORQA or separate service?
I like small independent services and don’t like when a single service do everything.

**Comment by mjankauskas** [comment]
App has to have this data before starting the service ![0457aab7-ad13-4d4a-84eb-54152299f43b (1).png](https://images.zenhubusercontent.com/5c3b439554281239e265e8e9/95390504-ad88-4465-ba79-0a0c2dff7843)

**Comment by zolia** [comment]
I don't think we should go after maxmind naming. For us its not a user type but a `node_type`.

**Comment by zolia** [comment]
>Node autodetects "user_type" same ways as "country" and adds to Proposal

Here I agree with @soffokl, if changing single parameter in proposal would mean that I as a node operator will earn more, I will definitely will do it. 

It is discovery or morqa should set this `node_type` after receiving proposal from a node.

**Comment by zolia** [comment]
I think we should extend MORQA service, but do not use ELK as a backend. /location and /me requests should go to different storage or service as a backend. We need to decide with storage we will be using.. I vote not for ELK since there are more straightforward ways to store / retrieve data. Also price of scaling will be too high, also ELK is not intended to be used as real-time database which is needed for us.

**Comment by zolia** [comment]
calls to /me /location by node should be optional IMHO. That is they can and they will fail. This should not prevent node from operating. We may build in redundancy, but even those can fail. All non essential endpoints should be optional.

**Comment by zolia** [comment]
>Node autodetects "user_type" same ways as "country" and adds to Proposal

@soffokl , we had a short discussion and it looks like best way to proceed is to pass this field from node runner into proposal. Then consumer will decide if he will trust it. Consumer may also try to validate this information by additionally checking with maxdb, morqa or any other means.
